### PR TITLE
Settings toggle to invert notification messages

### PR DESCRIPTION
**Pull request by jcarter3**
Telegram is the only chat application that inverts the order of messages in the notifications. I added an option in the notification settings to allow the user to select which order they want their messages to appear.

### CODE REVIEW

**Comment by jknigga** [comment]
Would really love to see this merged. The inverted message order in the notifications are really confusing.

**Comment by Bubu** [comment]
I'd like to merge this over at https://github.com/slp/Telegram-FOSS. Would this be okay with you?

If I read this correctly this defaults to the old behavior of having an inverted message order?

**Comment by jcarter3** [comment]
Sure, that's fine. Yes, it defaults to the current behavior.

**Comment by joshery420** [comment]
Please merge this @DrKLO. The notification order drives me nuts and no other Android app I've used displays notifications like Telegram does currently.

**Comment by conectapb** [comment]
@Bubu @DrKLO @jknigga  how do I change the colors of the layout already looked into several files and can not change

**Comment by joshery420** [comment]
@conectapb That might be a valid question, but it's unrelated to this PR. You should either post on a relevant Issue or make a new one.

**Comment by ADTC** [comment]
point.
inverted to make a 
Yes I wrote this
Thanks.
upside down?
read the messages in the notification 
Why would anyone in the right mind want to 
be considered a cosmetic bug and be fixed.
The current behavior should simply 
be a toggle setting.
this shouldn't even 
To be honest, 
**Please merge this.**

File: TMessagesProj/src/main/java/org/telegram/messenger/NotificationsController.java [hunk 1]
```diff
@@ -418,7 +418,11 @@ public void run() {
                             popupArray.add(0, messageObject);
                         }
                         delayedPushMessages.add(messageObject);
```
Comment (ADTC) [inline TMessagesProj/src/main/java/org/telegram/messenger/NotificationsController.java]: What about this line? Do you also need to add here in the correct order?

**Comment by jcarter3** [comment]
You might be right, I'll have to take a look. It seems to do different notifications based on regular vs group convos so I'll need to check the source closer.

**Comment by ADTC** [comment]
conflicts.
Now there are

**Comment by jcarter3** [comment]
I'll take a look and update it

**Comment by jknigga** [comment]
Really though, this is quite an easy fix. Please, please merge this.

**Comment by g1hostsilvestre** [comment]
@ADTC @Bubu @DrKLO 
how to add "lib" library to Apk
when I generate the apk here in the studio android it does not appear in the apk

**Comment by Royal2000H** [comment]
This is a must.

**Comment by Special68** [comment]
[#1400](https://github.com/DrKLO/Telegram/pull/1400)

**Comment by stackprogramer** [comment]
i finally build apk telegram  project thanks

**Comment by SHi-ON** [comment]
It seems to be an essential migration through GCM to FCM in the near future. according Google:

> Firebase Cloud Messaging (FCM) is the new version of GCM. It inherits the reliable and scalable GCM infrastructure, plus new features! See the FAQ to learn more. If you are integrating messaging in a new app, start with FCM. GCM users are strongly recommended to upgrade to FCM, in order to benefit from new FCM features today and in the future. 

That's could be a good news from @DrKLO if you make the switch!

**Comment by iM4hdi** [comment]
Sorry guys!
any news about iOS source code update?
did they stopped updating it?
@DrKLO

**Pull request closed without merging**